src/Bicep.Core/TypeSystem/DeployTimeConstantVisitor.cs

```diff
@@ -22,10 +22,12 @@ public sealed class DeployTimeConstantVisitor : SyntaxVisitor
         private SyntaxBase? errorSyntax;
         private string? currentProperty;
         private string? accessedSymbol;
+        private SyntaxBase? accessedSyntax;
+        private bool withinDeployTimeConstantScope;
```

Comment:
Any ideas on simplifying some of the state in this visitor? We already have a scary amount of individual fields, and bugs result in totally bizarre behavior like #1699 - so I'm a little hesitant to add more.

Reply:
Not sure how many state fields we can remove without a big refactoring, but I managed to get rid of one (merged `string accessedSymbol` and `SyntaxBase? accessedSyntax` into `DeclaredSymbol? referencedSymbol`).